### ISSUE TITLE
Feature request: ability to update text on save (without receiving focus)

### ISSUE THREAD

**Issue by chr15m** [asamuzaK/withExEditor#55]
In live-coding environments it is sub-optimal to switch back to the browser window after each save.

**Comment by asamuzaK** [comment]
Please explain in more detail why this function is required.

It is possible from the host side to observe temporary file changes using Node.js fs.watch().
However, temporary files are deleted when the browser is closed, so they will remain until then.
In other words, even if a temporary file becomes unnecessary, that file will still be observed, which is not efficient.

**Comment by asamuzaK** [comment]
For a note.
Another thing that is needed to consider carefully is that if this function is implemented, it will update the textarea of background tabs without any notification to the user.
Of course it is possible to notify to the user on attempt to sync contents, but it will be a very annoying behavior for the user?

**Comment by chr15m** [comment]
@asamuzaK I'm sorry for my lack of clarity.

The feature is to send changes to the browser as soon as the editor saves rather than requiring the user to switch focus back to the textarea itself. Where this is useful is in live-coding environments like this:

http://wavepot.com/

The user would like their changes to go quickly from the editor to the textarea and often in the background so that they can hear the changes without switching back and forward from the editor to the textarea.

https://www.youtube.com/watch?v=d-_J2zLtbtg

Agree there are complex UI considerations. Perhaps there could be a configuration option to enable this behaviour?

Thank you for your time and effort spent building this cool software!

**Comment by asamuzaK** [comment]
Thanks for clarify.
There are several things to consider.
So, it may take some time to fix this.

**Comment by asamuzaK** [comment]
options.html UI mock up:

en:
```
[ ] Automatically synchronize edited content on specific sites
    Sites (enter URL per line):
     ----------------------
    |textarea              |
    |                      |
    |                      |
     ----------------------

```

ja:
```
[ ] 特定のサイトでは編集したコンテンツを自動的に同期する
    同期するサイト（1行ごとにURLを入力）：
     ----------------------
    |textarea              |
    |                      |
    |                      |
     ----------------------
```

* Enable only for the sites specified by the user.
* No notification instead.

Thoughts?

**Comment by chr15m** [comment]
This would make it easier to work with live coding environments. :+1:

**Comment by asamuzaK** [comment]
Can you please give me additional info?

On sites using online editors such as CodePen and JSFiddle, the contents of temporary files are empty when selecting text editing even if there is a code written.
Both site seem to be using CodeMirror.

http://wavepot.com/ seems to be using the Ace editor, but as far as I tried, an empty temporary file was created just like CodeMirror.

When you are using it with withExEditor, is it working as expected?

I also tested with [Ace - The High Performance Code Editor for the Web](https://ace.c9.io/ "Ace - The High Performance Code Editor for the Web"), and I confirmed similar but a bit differnt issue.
If the name of the temporary file is like `textarea_0.txt`, an empty file is displayed as in CodeMirror.
But sometimes (depending on the place where I clicked?), temporary file like `div_7.txt` is created and I got the contents of a div element then.
However, code formatting is broken.

Related issue #56

**Comment by asamuzaK** [comment]
Try build.
[withExEditor v4.1.0a1](https://addons.mozilla.org/en-US/firefox/addon/withexeditor/versions/beta)

Note that host needs update too.
[Releases · asamuzaK/withExEditorHost](https://github.com/asamuzaK/withExEditorHost/releases "Releases · asamuzaK/withExEditorHost")

**Comment by asamuzaK** [comment]
Fixed #56 
It should now work in http://wavepot.com/

Another try build.
[withExEditor v4.1.0a3](https://addons.mozilla.org/en-US/firefox/addon/withexeditor/versions/beta "withExEditor :: Versions :: Add-ons for Firefox")

**Comment by asamuzaK** [comment]
Mac and Linux tester wanted.

@ssokolow @wavexx @Celti @ariel-co @maxigaz @braham-snyder

Hi,
If you still use withExEditor, can you help testing this new auto update feature?

I tested with my Windows 10 on
* https://ace.c9.io/
* https://codepen.io/pen/
* https://jsfiddle.net/

It did not work well on jsfiddle though.

Thanks.

**Comment by asamuzaK** [comment]
> It did not work well on jsfiddle though.

Fixed it.

**Comment by asamuzaK** [comment]
Released v4.1.0

**Comment by chr15m** [comment]
Excellent, thanks! 👍
